File: common/src/main/java/net/impleri/playerskills/integration/kubejs/skills/SpecializedSkillJS.java
package net.impleri.playerskills.integration.kubejs.skills;

import net.impleri.playerskills.api.TeamMode;
import net.impleri.playerskills.variant.specialized.SpecializedSkill;
import net.minecraft.resources.ResourceLocation;

public class SpecializedSkillJS extends SpecializedSkill {
    public SpecializedSkillJS(Builder builder) {
        super(builder.id, builder.options, builder.initialValue, builder.description, builder.changesAllowed, builder.teamMode, builder.notify, builder.notifyKey);
    }

    public static class Builder extends GenericSkillBuilderJS<String> {

        public Builder(ResourceLocation name) {
            super(name);
        }

        @Override
        public SpecializedSkillJS createObject() {
            return new SpecializedSkillJS(this);
        }

        public Builder splitEvenlyAcrossTeam() {
            teamMode = TeamMode.splitEvenly();

            return this;
        }
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/integrations/kubejs/skills/SpecializedSkillJS.kt
package net.impleri.playerskills.integrations.kubejs.skills

import net.impleri.playerskills.api.Skill
import net.impleri.playerskills.api.TeamMode
import net.impleri.playerskills.skills.specialized.SpecializedSkill
import net.minecraft.resources.ResourceLocation

class SpecializedSkillJS(builder: Builder) : SpecializedSkill(
  builder.id,
  builder.initialValue,
  builder.description,
  builder.options,
  builder.changesAllowed,
  builder.teamMode,
  builder.notify,
  builder.notifyKey,
) {
  class Builder(name: ResourceLocation) : GenericSkillBuilderJS<String>(name) {
    override fun createObject(): Skill<String> {
      return SpecializedSkillJS(this)
    }

    fun splitEvenlyAcrossTeam(): Builder {
      teamMode = TeamMode.splitEvenly()
      return this
    }
  }
}
